### PR TITLE
Add trailer

### PR DESCRIPTION
**Pull request by duarte**
Adds the play trailer button, and overlay, plus JS to toggle the overlay on and off.

**** Missing: the actual youtube embed, which should autoplay when the trailer is shown. cc/ @gidsg  ****

### CODE REVIEW

File: src/js/text/main.html [hunk 1]
```diff
@@ -33,6 +33,11 @@
                 </div>
                 <div class="docs--meta docs--standfirst-read-more">Full description and credits</div>
             </div>
+            <div class='docs--actions'>
+              <div class='docs--actions__trailer docs__shows-trailer'>
+                <span class='docs--actions__trailer__duration'>1:08</span>
```
Comment (akash1810) [inline src/js/text/main.html]: should this be coming from a YouTube API call like we do with the main video?

File: src/js/main.js [hunk 2]
```diff
@@ -42,6 +42,24 @@ export function init(el, context, config) {
             hideAboutBtn.classList.remove('docs--show-about');
         };
 
+
+        // Show the trailer on click #docs__playTrailer
+        const $showTrailerAll = builder.querySelectorAll('.docs__shows-trailer');
```
Comment (akash1810) [inline src/js/main.js]: we don't prefix variables with `$` elsewhere - we probably want to agree on a variable name style and be consistent across the project

File: src/js/text/main.html (hunk 1 above)
Comment (gidsg) [inline src/js/text/main.html]: Yes, good shout @akash1810 I think we should integrate that in a subsequent PR.

Comment (duarte) [inline src/js/text/main.html]: Yep that'd make sense — @gids is incorporating the actual youtube video in a later PR

File: src/js/main.js [hunk 3]
```diff
@@ -42,6 +42,24 @@ export function init(el, context, config) {
...
+        [].forEach.call($showTrailerAll, function($showTrailer) {
```
Comment (akash1810) [inline src/js/main.js]: are we going to have more than one trailer on a page? if not, we could simplify this by removing the `forEach` in favour of using a single element

Comment (duarte) [inline src/js/main.js]: Cool, I have updated the open showTrailer section — but left the closeTrailer section as a forEach as there are multiple elements that close the trailer?

**Pull request merged**